File: sdks/ruby-mcp/test/widget_test.rb
# frozen_string_literal: true

require "minitest/autorun"
require "solvapay/mcp"

class WidgetTest < Minitest::Test
  def test_default_widget_has_root_mount
    html = SolvaPay::Mcp.default_mcp_app_html
    assert_includes html, 'id="root"'
  end

  def test_widget_mime_type
    assert_equal "text/html;profile=mcp-app", SolvaPay::Mcp::MCP_APP_MIME_TYPE
  end

  def test_vendored_widget_matches_canonical
    canonical = File.expand_path("../../../tools/mcp-app-widget/mcp-app.html", __dir__)
    assert_equal File.read(canonical, encoding: "UTF-8"), SolvaPay::Mcp.default_mcp_app_html
  end
end

File: sdks/go/mcp/widget_test.go
package mcp

import (
	"os"
	"path/filepath"
	"runtime"
	"testing"
)

func TestDefaultMCPAppHTMLHasRootMount(t *testing.T) {
	html := DefaultMCPAppHTML()
	if html == "" || !defaultWidgetHasRootMount() {
		t.Fatal("vendored mcp-app.html is missing id=\"root\"")
	}
	if MCPAppMIMEType != "text/html;profile=mcp-app" {
		t.Fatalf("unexpected mime type %q", MCPAppMIMEType)
	}
}

func TestVendoredWidgetMatchesCanonical(t *testing.T) {
	_, thisFile, _, ok := runtime.Caller(0)
	if !ok {
		t.Fatal("runtime.Caller failed")
	}
	canonical := filepath.Join(filepath.Dir(thisFile), "..", "..", "..", "tools", "mcp-app-widget", "mcp-app.html")
	want, err := os.ReadFile(canonical)
	if err != nil {
		t.Fatal(err)
	}
	if DefaultMCPAppHTML() != string(want) {
		t.Fatal("go mcp-app.html drifted from tools/mcp-app-widget/mcp-app.html")
	}
}
